File: app/src/main/java/org/thoughtcrime/securesms/animation/AnimationRepeatListener.java
package org.thoughtcrime.securesms.animation;

import android.animation.Animator;

import androidx.annotation.NonNull;
import androidx.core.util.Consumer;

public final class AnimationRepeatListener implements Animator.AnimatorListener {

  private final Consumer<Animator> animationConsumer;

  public AnimationRepeatListener(@NonNull Consumer<Animator> animationConsumer) {
    this.animationConsumer = animationConsumer;
  }

  @Override
  public final void onAnimationStart(Animator animation) {
  }

  @Override
  public final void onAnimationEnd(Animator animation) {
  }

  @Override
  public final void onAnimationCancel(Animator animation) {
  }

  @Override
  public final void onAnimationRepeat(Animator animation) {
    this.animationConsumer.accept(animation);
  }
}

File: app/src/main/java/org/thoughtcrime/securesms/animation/AnimationRepeatListener.kt
/*
 * Copyright 2024 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.thoughtcrime.securesms.animation

import android.animation.Animator
import java.util.function.Consumer

class AnimationRepeatListener(private val animationConsumer: Consumer<Animator>) : Animator.AnimatorListener {
  override fun onAnimationStart(animation: Animator) = Unit
  override fun onAnimationEnd(animation: Animator) = Unit
  override fun onAnimationCancel(animation: Animator) = Unit

  override fun onAnimationRepeat(animation: Animator) {
    animationConsumer.accept(animation)
  }
}
